invenio_checks/components.py

```diff
@@ -21,31 +24,59 @@ def enabled(self):
         """Return if checks are enabled."""
         return current_app.config.get("CHECKS_ENABLED", False)
 
-    def update_draft(self, identity, data=None, record=None, errors=None, **kwargs):
+    def _run_checks(self, identity, data=None, record=None, errors=None, **kwargs):
         """Update handler."""
         errors = errors or []
 
         if not self.enabled:
             return
 
         communities = []
-        if record.review.is_open:
-            communities.append(record.review.community)
+        if record.parent.review and (
+            record.parent.review.status == "submitted"
+            or record.parent.review.status == "created"
+        ):
+            # drafts can only be submitted to one community
+            communities.append(record.parent.review.receiver.resolve().id)
+        else:
+            results = current_requests_service.search(
+                identity,
+                extra_filter=dsl.query.Bool(
+                    "must",
+                    must=[
+                        dsl.Q("term", **{"topic.record": record.pid.pid_value}),
+                        dsl.Q("term", **{"is_open": True}),
+                    ],
+                ),
+            )
 
-        communities.extend(record.parent.communities.entries)
-        checks = itertools.chain(*[c.checks for c in communities])
+            if results.total > 0:
+                for result in results:
+                    communities.append(result.get("receiver", {}).get("community"))
+            else:
+                return
 
-        for check in checks:
+        all_checks = [
+            (CheckConfig.query.filter(CheckConfig.community_id == c).all(), c)
+            for c in communities
+        ]
+
+        for checks, community in all_checks:
             try:
-                res = check.run(data, record)
-                if not res.sync:
-                    continue
-                for error in res.errors:
-                    errors.append(error)
+                for check in checks:
+                    check_cls = current_checks_registry.get(check.check_id)
+                    res = check_cls().run(record, check.params, community)
+                    if not res.sync:
+                        continue
+                    for error in res.errors:
+                        errors.append(*error)
```

Comment:
```suggestion
                    errors.extend(res.errors)
```

nit/minor: although I'm also curious why we need the `*error`,  which would also fail if there are none or more than one elements in `error`.

Reply:
If you use extend you get something like this
```
16:31:13.88 .... errors = [{'field': 'metadata.title', 'messages': ['Missing data for required field.']},
16:31:13.88                None,
16:31:13.88                None,
16:31:13.88                [{'context': {'check': 'metadata',
16:31:13.88                              'community': UUID('1f6910f0-80aa-477c-baa5-3d176e0237f3'),
16:31:13.88                              'rule': 'eu-funding'},
16:31:13.88                  'field': 'metadata.funding',
16:31:13.88                  'messages': ['Records must have at least one EU-funded project'],
16:31:13.88                  'severity': 'info'}]]
```

Reply:
nvm I fixed it to all use extend